botocore/signers.py

```diff
@@ -196,6 +208,65 @@ def generate_presigned_url(self, request_dict, expires_in=3600,
         return request.url
 
 
+class CloudFrontSigner(object):
+
+    @staticmethod
+    def b64encode(content):
+        return base64.b64encode(content).translate(maketrans(b'+=/', b'-_~'))
+
+    def __init__(self, key_id, private_key):
+        """Create a CloudFrontSigner.
+
+        :type key_id: str
+        :param key_id: The CloudFront Key Pair ID
+
+        :type private_key: str
+        :param private_key: The private key string used for signing.
+        """
+        self.key_id = key_id
+        self.rsa_pk = rsa.PrivateKey.load_pkcs1(private_key.encode('utf8'))
+
+    def sign(self, url, expires, starts=None, ip_address=None):
```

Comment:
CloudFront supports canned policies and custom policies. The way this is currently implemented does not support custom policies. I think you should only use a canned policy if the user does not supply a custom policy. Perhaps there could be two methods -- one that signs with a custom policy, and one that accepts a few abstracted parameters like expires, creates a policy, and then calls the other function that signs custom policies. Here's how this same functionality is implemented in the PHP SDK: https://github.com/aws/aws-sdk-php/blob/master/src/CloudFront/UrlSigner.php#L61